modules/rest/src/main/java/com/enonic/xp/app/contentstudio/rest/resource/content/ContentResource.java

```diff
@@ -794,11 +789,11 @@ private ContentIds problematicDependantsOnTop( final ContentIds dependentContent
     private ContentIds filterOutStatus( final ContentIds contentIds, final CompareContentResults compareResults,
                                         final CompareStatus status )
     {
-        return ContentIds.from( compareResults.stream()
-                                    .filter( result -> result.getCompareStatus() != status )
-                                    .map( CompareContentResult::getContentId )
-                                    .filter( contentIds::contains )
-                                    .collect( Collectors.toList() ) );
+        final ContentIds unorderedIdsWithoutStatus = ContentIds.from( compareResults.stream()
```

Comment:
Well, not quite. `filterOutStatus()` is just a common method, created to avoid duplication in `getNotPendingDeletion()` and `getNotEqual()`, that are both utility methods, accepting CompareResult, that is different in the context they are called from.
`getNotEqual()` is used in in new relatevely new `getOutboundDependenciesIds()` method, where `contentService.compare()` is passed.

Reply:
`PendingDelete` - is deprecated status and the related code can be removed from the CS.
 different context of the calls - is the problem. Passed `CompareResults` contains more data than `contentIds` and its not obvious from the method signature what should be filtered from what.